File: src/dataset/c/armstrong-numbers_exercism-GDYendell/claude/ClaudeSonnet4_armstrong_numbers_v2.c
#include "armstrong_numbers.h"

bool is_armstrong_number(int candidate) {
    if (candidate < 0) return false;
    
    int original = candidate;
    int len = 0;
    int temp = candidate;
    
    // Count digits
    do {
        len++;
        temp /= 10;
    } while (temp > 0);
    
    // Calculate armstrong sum
    int armstrong_number = 0;
    temp = candidate;
    while (temp > 0) {
        int digit = temp % 10;
        int power = 1;
        for (int i = 0; i < len; i++) {
            power *= digit;
        }
        armstrong_number += power;
        temp /= 10;
    }
    
    return original == armstrong_number;
}

File: src/dataset/Java/armstrong-numbers_exercism-uzilan/claude/ClaudeSonnet4_ArmstrongNumbers_v2.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        if (numberToCheck < 0) return false;
        
        int original = numberToCheck;
        int numDigits = 0;
        int temp = numberToCheck;
        
        // Count digits
        while (temp > 0) {
            numDigits++;
            temp /= 10;
        }
        
        // Calculate sum of powers
        int sum = 0;
        temp = numberToCheck;
        while (temp > 0) {
            int digit = temp % 10;
            sum += pow(digit, numDigits);
            temp /= 10;
        }
        
        return sum == original;
    }
    
    private int pow(int base, int exp) {
        int result = 1;
        for (int i = 0; i < exp; i++) {
            result *= base;
        }
        return result;
    }
}